File: ProPanel/frontend/src/api/dns.ts
import axios from "axios";
import { getToken } from "./auth";

export async function getZones() {
  const resp = await axios.get("/api/v1/dns/zones", {
    headers: { Authorization: `Bearer ${getToken()}` },
  });
  return resp.data;
}

export async function getRecords(zone_id: string) {
  const resp = await axios.get("/api/v1/dns/records", {
    headers: { Authorization: `Bearer ${getToken()}` },
    params: { zone_id }
  });
  return resp.data;
}

export async function addRecord(zone_id: string, type: string, name: string, content: string) {
  const resp = await axios.post(
    "/api/v1/dns/add",
    { zone_id, type, name, content },
    { headers: { Authorization: `Bearer ${getToken()}` } }
  );
  return resp.data;
}

export async function deleteRecord(zone_id: string, record_id: string) {
  const resp = await axios.post(
    "/api/v1/dns/delete",
    { zone_id, record_id },
    { headers: { Authorization: `Bearer ${getToken()}` } }
  );
  return resp.data;
}

File: ProPanel/backend/app/api/api_v1/endpoints/dns.py
from fastapi import APIRouter, Depends, HTTPException
from app.api import deps
from typing import List
import os

import CloudFlare

router = APIRouter()

CF_EMAIL = os.environ.get("CF_EMAIL")
CF_TOKEN = os.environ.get("CF_TOKEN")

def get_cf():
    return CloudFlare.CloudFlare(email=CF_EMAIL, token=CF_TOKEN)

@router.get("/zones")
def list_zones(current_user=Depends(deps.get_current_active_superuser)):
    cf = get_cf()
    return cf.zones.get()

@router.get("/records")
def list_records(zone_id: str, current_user=Depends(deps.get_current_active_superuser)):
    cf = get_cf()
    return cf.zones.dns_records.get(zone_id)

@router.post("/add")
def add_record(
    zone_id: str,
    type: str,
    name: str,
    content: str,
    current_user=Depends(deps.get_current_active_superuser),
):
    cf = get_cf()
    rec = {
        "type": type.upper(),
        "name": name,
        "content": content,
        "ttl": 120,
        "proxied": False,
    }
    cf.zones.dns_records.post(zone_id, data=rec)
    return {"ok": True}

@router.post("/delete")
def delete_record(zone_id: str, record_id: str, current_user=Depends(deps.get_current_active_superuser)):
    cf = get_cf()
    cf.zones.dns_records.delete(zone_id, record_id)
    return {"ok": True}
